### PR TITLE
DEV: Fix failing chat spec and add unexpected failure indicator

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit fixes the UpdateUserLastRead spec which was checking
for a message ID that did not exist -- this could fail at times
since message ID 2 could exist. Better to create + destroy a message
since then it's guaranteed we have a unique ID.

This also attempts to clarify a step that we expect to fail which
succeeds instead by adding another emoji next to the success tick,
to try make it even clearer for the operator which step has the problem.

Also removes some uses of unless in Services::Base, we generally prefer
to use alternatives, since unless can be hard to parse in a lot of
cases.

Before

image

After

image

### CODE REVIEW

File: plugins/chat/lib/steps_inspector.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def type
       end
 
       def emoji
-        return "❌" if failure?
+        return spec_fail_unexpected_result? ? "❌ 🚧" : "❌" if failure?
```
Comment (Flink) [inline plugins/chat/lib/steps_inspector.rb:33]: This actually won’t work as the example shown in the screenshots (it should be applied to ✅ too) :thinking:

File: plugins/chat/spec/support/chat_service_matcher.rb [hunk 2]
```diff
@@ -23,6 +23,9 @@ def failure_message
           else
             "expected the service to fail but it succeeded."
           end
+
+        result[step].merge("spec.fail_step.unexpected_result" => true) if !step_failed?
```
Comment (Flink) [inline plugins/chat/spec/support/chat_service_matcher.rb:27]: I already played a lot with this code and it would be simpler to apply this each time `failure_message` is called because if it’s called, then we’re in a state we didn’t expect :slightly_smiling_face: 
And we should use it in `failure_message_when_negated`. We need to update the inspector specs too.

**Comment by Flink** [comment]
Forgot a comment: I don’t really understand what value `if !condition` brings over `unless condition`. To me, the latter is quicker to understand as it reads as plain English. I agree that abusing `unless` is not a good thing to do and for example `unless … else … end` blocks should be avoided. But for oneliners/simple conditions, I don’t really see the problem.

**Review by jjaffeux — APPROVED** [review]

File: plugins/chat/app/services/base.rb [hunk 3]
```diff
@@ -197,7 +197,7 @@ def initialize(name, method_name = name, class_name: nil)
         def call(instance, context)
           method = instance.method(method_name)
           args = {}
-          args = context.to_h unless method.arity.zero?
+          args = context.to_h if method.arity.positive?
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply plugins/chat/app/services/base.rb]: Thanks didn't realise that!
